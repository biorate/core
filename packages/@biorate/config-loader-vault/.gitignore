# compiled output
dist
node_modules
cache
keys

# Logs
logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
lerna-debug.log*

# Diagnostic reports (https://nodejs.org/api/report.html)
report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json

# OS
.DS_Store

# Coverage directory used by tools like istanbul
coverage
*.lcov

# nyc test coverage
.nyc_output

# Optional npm cache directory
.npm

# Optional eslint cache
.eslintcache

# IDEs and editors
.idea
.project
.classpath
.c9/
*.launch
.settings/
*.sublime-workspace

# IDE - VSCode
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json

# Package lock
package-lock.json
