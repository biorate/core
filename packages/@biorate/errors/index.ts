import { format } from 'util';

/**
 * @description
 * This module provides a base class for creating error classes with error code (class name),
 * message template, meta data support out of the box.
 *
 * @example
 * ```ts
 * import { BaseError } from '@biorate/errors';
 *
 * export class MyAwesomeError extends BaseError {
 *   constructor(args?: any[], meta?: any) {
 *     super(`Oops... Some error happen, at [%s], in [%s]`, args, meta);
 *   }
 * }
 *
 * const e = new MyAwesomeError([new Date(), 'core'], { hello: 'world!' });
 *
 * console.log(e.meta); // { hello: 'world!' }
 * console.log(e.code); // MyAwesomeError
 *
 * throw e;
 *           ^
 * MyAwesomeError: Oops... Some error happen, at [2021-05-13T09:19:22.511Z], in [core]
 *     at Object.<anonymous> (..core/packages/@biorate/errors/index.ts:28:11)
 *     at Module._compile (internal/modules/cjs/loader.js:1138:30)
 * ```
 */
export class BaseError extends Error {
  #meta: any = null;

  public constructor(
    message: string,
    args?: unknown[],
    meta?: unknown,
    ...options: unknown[]
  ) {
    // @ts-ignore: { cause } property added in 16.9.0
    super(format(message, ...(args || [])), ...options);
    this.#meta = meta;
    this.name = this.constructor.name;
    Error.captureStackTrace(this, this.constructor);
  }

  public get meta() {
    return this.#meta;
  }

  public get code() {
    return this.constructor.name;
  }
}
