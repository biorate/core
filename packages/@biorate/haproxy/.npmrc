registry=https://art.lmru.tech/api/npm/npm-payment/
package-lock=false
