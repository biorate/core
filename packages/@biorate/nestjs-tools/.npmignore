/src
!/src/enums/*
/index.ts
