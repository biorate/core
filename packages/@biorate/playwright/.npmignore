/tests
/allure-results
/coverage
/.nyc_output
