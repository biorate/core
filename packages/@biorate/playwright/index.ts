export * from './interfaces';
export * from './src';
