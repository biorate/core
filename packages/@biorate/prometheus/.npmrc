registry=https://art.lmru.tech/api/npm/npm-payment/
