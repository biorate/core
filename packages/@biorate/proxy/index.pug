doctype html
html(lang="en")
  head
    title= "Proxy stats"
  style.
    table {
      width: 100%;
      border-collapse: collapse;
      border: 2px solid rgb(140 140 140);
      font-family: sans-serif;
      font-size: 0.8rem;
      letter-spacing: 1px;
    }

    caption {
      padding: 10px;
      font-weight: bold;
      font-size: 16px;
    }

    thead,
    tfoot {
      background-color: rgb(228 240 245);
    }

    th,
    td {
      border: 1px solid rgb(160 160 160);
      padding: 8px 10px;
    }

    th[scope=row] {
      font-weight: 100;
    }

    td:last-of-type {
      text-align: center;
    }

    tbody > tr:nth-of-type(even) {
      background-color: rgb(237 238 242);
    }

    tfoot th {
      text-align: center;
    }

    tfoot td {
      font-weight: bold;
    }

    .active {
      background: rgba(173, 255, 47, 0.35);
    }

    .inactive {
      background: rgba(255, 0, 0, 0.30);
    }
  body
    table
      caption Proxy stats
      thead
        tr
          th(scope="col") Connection
          th(scope="col") Proxy host
          th(scope="col") Client host
          th(scope="col") Active
          th(scope="col") Writed
          th(scope="col") Readed
      tbody
        each row in rows
          tr
            th(scope="row") #{row.connection}
            th(scope="row") #{row.proxyHost}
            th(scope="row") #{row.clientHost}
            th(scope="row" class=row.active ? 'active' : 'inactive') #{row.active}
            th(scope="row") #{row.writed}
            th(scope="row") #{row.readed}
      tfoot
        tr
          th(scope="col") Connection
          th(scope="col") Proxy host
          th(scope="col") Client host
          th(scope="col") Active
          th(scope="col") Writed
          th(scope="col") Readed
