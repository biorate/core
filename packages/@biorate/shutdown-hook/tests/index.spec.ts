import { use, expect } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';
import { fork } from 'child_process';
import { path } from '@biorate/tools';

use(jestSnapshotPlugin());

describe('shutdown-hook', function () {
  this.timeout(10000);

  for (const file of [
    'process.EXIT',
    'process.SHUTDOWN',
    'process.SIGINT',
    'process.SIGTERM',
  ])
    it(file, (done) => {
      const child = fork(path.create(__dirname, '__mocks__', `${file}.ts`), [], {
        execArgv: ['-r', 'ts-node/register'],
        stdio: 'overlapped',
      });
      child.stdout?.on?.('data', (data) => {
        expect(data.toString()).toMatchSnapshot();
        done();
      });
    });
});
