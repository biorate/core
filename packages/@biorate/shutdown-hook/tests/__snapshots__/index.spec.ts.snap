// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shutdown-hook process.EXIT 1`] = `
"EXIT
"
`;

exports[`shutdown-hook process.SHUTDOWN 1`] = `
"SHUTDOWN
"
`;

exports[`shutdown-hook process.SIGINT 1`] = `
"SIGINT
"
`;

exports[`shutdown-hook process.SIGTERM 1`] = `
"SIGTERM
"
`;
